packages/babel-plugin/src/clone-element/index.ts

```diff
@@ -0,0 +1,183 @@
+    t.isExpression(path.node.arguments[0]) &&
+    path.scope.hasOwnBinding(path.node.callee.name)
+  ) {
+    const binding = path.scope.getBinding(path.node.callee.name)?.path.node;
+
+    if (
+      !!resolveIdentifierComingFromDestructuring({ name: 'css', node: binding as t.Expression })
+    ) {
+      // c({}) rename call
+      const styles = path.node.arguments as t.Expression[];
+      return styles;
+    }
+  }
+
+  if (t.isCallExpression(path.node) && t.isMemberExpression(path.node.callee)) {
+    if (
+      t.isIdentifier(path.node.callee.property) &&
+      path.node.callee.property.name === 'css' &&
+      t.isExpression(path.node.arguments[0])
+    ) {
+      // props.css({}) call
+      const styles = path.node.arguments as t.Expression[];
+      return styles;
+    }
+  }
+
+  if (t.isTaggedTemplateExpression(path.node)) {
+    const styles = path.node.quasi;
+    return styles;
+  }
+
+  return undefined;
+};
+
+/**
+ * Takes a React.cloneElement invocation and transforms it into a compiled component.
+ * This method will traverse the AST twice,
+ * once to replace all calls to `css`,
+ * and another to replace `style` usage.
+ *
+ * `React.cloneElement(<Component />, { css: {} })`
+ *
+ * @param path {NodePath} The opening JSX element
+ * @param meta {Metadata} Useful metadata that can be used during the transformation
+ */
+export const visitCloneElementPath = (path: NodePath<t.CallExpression>, meta: Metadata): void => {
+  // if props contains a `css` prop, we need to transform it.
+  const props = path.node.arguments[1];
+
+  if (props.type !== 'ObjectExpression') {
+    // TODO: handle this case properly
+    console.error('cloneElement props are not an ObjectExpression');
+    return;
+  }
+
+  const collectedVariables: CSSOutput['variables'] = [];
+  const collectedSheets: string[] = [];
+
+  // First pass to replace all usages of `css({})`
+  path.traverse({
+    CallExpression(path) {
+      const styles = extractStyles(path);
+
+      if (!styles) {
+        // Nothing to do - skip.
+        return;
+      }
+
+      const builtCss = buildCss(styles, meta);
+      const { sheets, classNames } = transformCssItems(builtCss.css, meta);
+
+      collectedVariables.push(...builtCss.variables);
+      collectedSheets.push(...sheets);
+
+      path.replaceWith(
+        t.callExpression(t.identifier(getRuntimeClassNameLibrary(meta)), [
+          t.arrayExpression(classNames),
+        ])
+      );
+
+      // find ancestor cloneElement callExpression
+      const ancestorPath = path.findParent(
+        (p) =>
+          p.isCallExpression() &&
+          t.isIdentifier(p.node.callee) &&
+          p.node.callee.name === 'cloneElement'
+      ) as NodePath<t.CallExpression>;
+
+      if (!ancestorPath) {
+        return;
+      }
+
+      ancestorPath.replaceWith(buildCompiledCloneElement(ancestorPath.node, builtCss, meta));
+    },
+  });
+
+  // // Second pass to replace all usages of `style`.
```

Comment:
TODO, makes sense to hook the `style` prop up as well

Reply:
Does anything else ever use `style` in Babel?  `style={css(…)}` isn't valid, right?

Reply:
That's only when it's in runtime mode, but yeah, maybe right…really glad if you can learn this part of it because I have no clue how extraction or runtime _actually work_ 🤣

Reply:
Neither hahah